internal/cli/main_test.go

```diff
@@ -0,0 +1,48 @@
+package cli
+
+import (
+	"testing"
+)
+
+func TestEscapeIfNeeded(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		expected string
+	}{
+		{
+			name:     "AlreadyEscaped",
+			input:    "hello%20world",
+			expected: "hello%20world",
```

Comment:
The flaw with this approach is a set of characters in a password like `%20` would not be interpreted correctly.